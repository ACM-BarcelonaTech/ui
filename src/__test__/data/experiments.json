[
  {
    "processingConfig": {},
    "lastViewed": null,
    "createdDate": "0000-00-00T00:00:00.000Z",
    "meta": {
      "organism": null,
      "type": "10x"
    },
    "sampleIds": [],
    "description": "Analysis description",
    "experimentId": "'testae48e318dab9a1bd0bexperiment'",
    "projectId": "mock594b-mock-test-ba89-c2c812d3test",
    "experimentName": "Test Project"
  }
]