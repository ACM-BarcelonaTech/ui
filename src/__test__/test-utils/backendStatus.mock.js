const backendStatus = {
  gem2: {
    executionArn: '"arn:aws:states:eu-east-1:000000000000:execution:biomage-gem2s-test-test9cd23b9138d3d5b6141835c85101495a92cd:fb047b29-a9fb-4e36-a0fc-b792e243test"',
    paramsHash: '67515f56ad338b8c7339593582614ff62e376587', // pragma: allowlist secret
    stateMachineArn: 'arn:aws:states:eu-east-1:000000000000:stateMachine:biomage-gem2s-test-test9cd23b9138d3d5b6141835c85101495atest',
  },
  pipeline: {
    executionArn: 'arn:aws:states:eu-west-1:000000000000:execution:biomage-qc-test-test9cd23b9138d3d5b6141835c85101495a92cd:aef9d56d-a20c-4c10-ac1b-2c37cb87test',
    stateMachineArn: 'arn:aws:states:eu-west-1:000000000000:execution:biomage-qc-test-test9cd23b9138d3d5b6141835c85101495a92cd:aef9d56d-a20c-4c10-ac1b-2c37cb87test',
    completedSteps: [
      'ClassifierFilter',
      'CellSizeDistributionFilter',
      'MitochondrialContentFilter',
      'NumGenesVsNumUmisFilter',
      'DoubletScoresFilter',
      'DataIntegration',
      'ConfigureEmbedding',
    ],
    startDate: '2021-08-16T12:10:32.320Z',
    stopDate: '2021-08-16T12:21:07.847Z',
    status: 'SUCCEEDED',
  },
  worker: {
    ready: true,
    restartCount: 0,
    started: true,
    status: 'Running',
  },
};

export default backendStatus;
