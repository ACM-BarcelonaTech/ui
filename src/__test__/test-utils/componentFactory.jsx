import React from 'react';
import _ from 'lodash';

const createTestComponentFactory = (Component, defaultProps) => (customProps = {}) => {
  const props = _.merge(
    defaultProps,
    customProps,
  );

  // eslint-disable-next-line react/jsx-props-no-spreading
  return <Component {...props} />;
};

export default createTestComponentFactory;
