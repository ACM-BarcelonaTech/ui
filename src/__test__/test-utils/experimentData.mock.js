import fake from '__test__/test-utils/constants';
import mockBackendStatus from '__test__/test-utils/backendStatus.mock';

const mockExperimentData = {
  data: {
    experimentId: fake.EXPERIMENT_ID,
    experimentName: fake.EXPERIMENT_NAME,
    meta: mockBackendStatus,
    projectId: fake.PROJECT_ID,
    sampleIds: [fake.SAMPLE_ID],
  },
};

export default mockExperimentData;
