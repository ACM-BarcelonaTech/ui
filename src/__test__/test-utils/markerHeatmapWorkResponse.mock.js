const workResponse = {
  data: {
    GENE1: {
      rawExpression: {
        expression: [1, 2, 3, 4, 5],
        mean: 3,
        stdev: 1.41421,
      },
      truncatedExpression: {
        expression: [1, 2, 3, 4, 5],
        min: 0,
        max: 5,
      },
      zScore: [0.1, 0.2, 0.3, 0.4, 0.5],
    },
    GENE2: {
      rawExpression: {
        expression: [1, 2, 3, 4, 5],
        mean: 3,
        stdev: 1.41421,
      },
      truncatedExpression: {
        expression: [1, 2, 3, 4, 5],
        min: 0,
        max: 5,
      },
      zScore: [0.1, 0.2, 0.3, 0.4, 0.5],
    },
    GENE3: {
      rawExpression: {
        expression: [1, 2, 3, 4, 5],
        mean: 3,
        stdev: 1.41421,
      },
      truncatedExpression: {
        expression: [1, 2, 3, 4, 5],
        min: 0,
        max: 5,
      },
      zScore: [0.1, 0.2, 0.3, 0.4, 0.5],
    },
  },
  order: ['GENE1', 'GENE2', 'GENE3'],
};

export default workResponse;
