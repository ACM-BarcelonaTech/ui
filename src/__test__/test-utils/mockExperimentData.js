import fake from '__test__/test-utils/constants';
import mockBackendStatus from '__test__/data/backend_status.json';

const generateMockExperimentData = (experimentId) => ({
  data: {
    experimentId,
    experimentName: `Experiment ${experimentId}`,
    meta: mockBackendStatus,
    projectId: fake.PROJECT_ID,
    sampleIds: [fake.SAMPLE_ID],
  },
});

export default generateMockExperimentData;
