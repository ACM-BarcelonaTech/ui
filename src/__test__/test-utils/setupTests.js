import '@testing-library/jest-dom/extend-expect';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import preloadAll from 'jest-next-dynamic';

configure({ adapter: new Adapter() });
jest.mock('localforage');
jest.mock('utils/pushNotificationMessage');

// Add stuff that needs to run once, before all tests
beforeAll(async () => {
  await preloadAll();
});

// Add stuff that needs to run before each test
beforeEach(async () => {
  console.log('HELLO WORLD ++++++++++');
});
