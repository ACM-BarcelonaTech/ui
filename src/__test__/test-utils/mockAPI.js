import _ from 'lodash';

// Defaul platform data
import cellSetsData from '__test__/data/cell_sets.json';
import backendStatusData from '__test__/data/backend_status.json';
import processingConfigData from '__test__/data/processing_config.json';
import mockExperimentData from '__test__/test-utils/experimentData.mock';

const promiseResponse = (
  response,
  options = {},
) => Promise.resolve(new Response(response, options));

const statusResponse = (code, body) => Promise.resolve({
  status: code,
  body: JSON.stringify(body),
});

const workerResponse = (body, error = false) => promiseResponse(JSON.stringify({
  results: [
    {
      body: JSON.stringify(body),
    },
  ],
  response: { error },
}));

const generateDefaultMockAPIResponses = (mockExperimentId) => ({
  [mockExperimentId]: () => promiseResponse(
    JSON.stringify(mockExperimentData),
  ),
  processingConfig: () => promiseResponse(
    JSON.stringify(processingConfigData),
  ),
  '/cellSets': () => promiseResponse(
    JSON.stringify(cellSetsData),
  ),
  '/backendStatus': () => promiseResponse(
    JSON.stringify(backendStatusData),
  ),
});

const mockApi = (apiMapping) => (req) => {
  const path = req.url;

  const key = _.find(
    Object.keys(apiMapping),
    (urlStub) => path.endsWith(urlStub),
  );

  return apiMapping[key](req);
};

export default mockApi;
export {
  generateDefaultMockAPIResponses,
  promiseResponse,
  statusResponse,
  workerResponse,
};
