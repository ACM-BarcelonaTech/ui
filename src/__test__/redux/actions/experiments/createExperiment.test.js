import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';

import { EXPERIMENTS_SAVING, EXPERIMENTS_CREATED, EXPERIMENTS_SAVED } from 'redux/actionTypes/experiments';
import { createExperiment } from 'redux/actions/experiments';
import initialExperimentState from 'redux/reducers/experiments/initialState';
import initialProjectState, { projectTemplate } from 'redux/reducers/projects/initialState';

import '__test__/test-utils/setupTests';

const mockStore = configureStore([thunk]);

enableFetchMocks();

describe('createExperiment', () => {
  const projectUuid = 'project-1';

  const mockState = {
    projects: {
      ...initialProjectState,
      [projectUuid]: {
        ...projectTemplate,
        uuid: projectUuid,
        experiments: [],
      },
    },
    experiments: {
      ...initialExperimentState,
    },
  };

  beforeEach(() => {
    const response = new Response(JSON.stringify({}));

    fetchMock.resetMocks();
    fetchMock.doMock();
    fetchMock.mockResolvedValueOnce(response);
  });

  it('Dispatches action when called', async () => {
    const store = mockStore(mockState);
    await store.dispatch(createExperiment(projectUuid));

    const actions = store.getActions();

    expect(actions[0].type).toEqual(EXPERIMENTS_SAVING);
    expect(actions[1].type).toEqual(EXPERIMENTS_CREATED);
    expect(actions[2].type).toEqual(EXPERIMENTS_SAVED);

    expect(actions[1].payload).toMatchSnapshot();
  });
});
