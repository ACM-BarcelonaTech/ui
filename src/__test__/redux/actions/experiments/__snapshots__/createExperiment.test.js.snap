// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createExperiment Dispatches action when called 1`] = `
Object {
  "experiment": Object {
    "createdDate": "2020-01-01T00:00:00.000Z",
    "description": "Analysis description",
    "id": "b3f6c0ca86ec045c84f380cd5016972e",
    "lastViewed": null,
    "meta": Object {
      "organism": null,
      "type": "10x",
    },
    "name": undefined,
    "notifyByEmail": true,
    "projectUuid": "project-1",
    "sampleIds": Array [],
  },
}
`;

exports[`createExperiment Dispatches action when called 2`] = `"{\\"description\\":\\"Analysis description\\",\\"createdDate\\":\\"2020-01-01T00:00:00.000Z\\",\\"lastViewed\\":null,\\"notifyByEmail\\":true,\\"meta\\":{\\"organism\\":null,\\"type\\":\\"10x\\"},\\"sampleIds\\":[],\\"experimentId\\":\\"b3f6c0ca86ec045c84f380cd5016972e\\",\\"projectId\\":\\"project-1\\"}"`;
